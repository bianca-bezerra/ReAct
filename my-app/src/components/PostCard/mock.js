export const mockPostCard = {
  id: '10',
  title: 'titulo',
  body: 'corpo do post',
  cover: 'fotinha',
  alt: 'alt de fotinha',
};
