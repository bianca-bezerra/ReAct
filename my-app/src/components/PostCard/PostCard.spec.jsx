import { render, screen } from "@testing-library/react"
import { PostCard } from "."
import { mockPostCard } from "./mock"

const props = mockPostCard

describe("<PostCard />", () => {
  it("should render PostCard correctly", () => {
    render(
      <PostCard {...props} />)
    expect(screen.getByRole('img', { name: props.alt })).toHaveAttribute('src', props.cover)
    expect(screen.getByRole('heading', { name: props.title })).toBeInTheDocument()
    expect(screen.getByText(props.body)).toBeInTheDocument()
  })

  it("should match PostCard snap", () => {
    const { container } = render(
      <PostCard {...props} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
