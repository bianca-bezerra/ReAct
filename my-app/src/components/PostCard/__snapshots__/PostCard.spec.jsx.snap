// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match PostCard snap 1`] = `
<div
  class="post"
>
  <div
    class="post-content"
  >
    <img
      alt="alt de fotinha"
      src="fotinha"
    />
    <h1>
      titulo
    </h1>
    <p>
      corpo do post
    </p>
  </div>
</div>
`;
